### PR TITLE
Full WebRTC Support

### LINKED ISSUE #6: Make hyperproxy node behave like a WebRTC node
Hyperproxy node should be able to take in WebRTC traffic, fetch the files from Dat, and then send them back via WebRTC

### LINKED ISSUE #33: Need to figure out how to utilize Simple Peer to send data directly to other nodes
We are running into an issue where we are not able to send data between peers via Simple Peer after receiving a data event from SignalHub. 

The code in question is here:

https://github.com/goonism/hyperproxy/blob/master/packages/hyperproxy-node/src/index.js#L52

This code performs the functionality that we would like it to, it's just that it's using the SignalHub as a centralized source to proxy requests to all the other nodes in the channel.

Instead, the node that receives the event should reply directly to the node that sent it the request.

### PR DESCRIPTION
**Pull request by louisgv**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #6 , #4  
| Patch: Bug Fix?          | Y
| Major: Breaking Change?  | Y
| Minor: New Feature?      | N
| Tests Added + Pass?      | N
| Documentation PR         | N 
| Any Dependency Changes?  | N
| License                  | MIT

@ColdSauce : Plx rebase, have this PR (https://github.com/mafintosh/webrtc-swarm/pull/33) reviewed on upstream, and refactor the code to remove comment and so on. But it's fully functional :p 

Closes #33 #31

### CODE REVIEW

**Comment by ColdSauce** [comment]
Awesome -- seems to work. However, it fails when we trying to request large files. We might need to do file partitioning (?)

File: packages/hyperproxy-browser/src/index.js [hunk 1]
```diff
@@ -19,6 +19,57 @@ function init() {
...
+    const originalOnMessage = client.hub.onMessage;
+
+    window.addEventListener('unload', ()=>{
+        client.hub.broadcast(activeHash, {
+            from: client.swarm.me,
+            type: HUB_MSG_TYPE.LEAVE
+        });
+    });
+
+    // Hack the onMessage event of signalhubws to receive data from Hub
```
Comment (ColdSauce) [inline packages/hyperproxy-browser/src/index.js]: Remove commented out code

File: packages/hyperproxy-browser/src/index.js [hunk 2]
```diff
@@ -19,6 +19,57 @@ function init() {
...
+
+        peer.on('data', function(data) {
+            console.log("got data from peer 2: ", data);
+            renderOutput(data);
+        });
+    });
+
+    function renderOutput(data) {
+        const {body, type, from} = JSON.parse(Buffer.from(data).toString('utf8'));
+        console.log(body);
```
Comment (ColdSauce) [inline packages/hyperproxy-browser/src/index.js]: remove log

Comment (ColdSauce) [reply packages/hyperproxy-browser/src/index.js]: ( or convert it to use logging framework )

File: packages/hyperproxy-config/index.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+process.env.LOCAL = true;
```
Comment (ColdSauce) [inline packages/hyperproxy-config/index.js]: Remove this

File: packages/hyperproxy-hub-client/src/index.js [hunk 4]
```diff
@@ -34,11 +34,16 @@ export default class HyperproxyHubClient {
             sw.close();
         });
 
+        sw.on('data', function(m) {
+            console.log(m);
```
Comment (ColdSauce) [inline packages/hyperproxy-hub-client/src/index.js]: should these logging statements be here?

File: packages/hyperproxy-hub/src/index.js [hunk 5]
```diff
@@ -31,22 +32,46 @@ function onConnection(ws) {
...
+
+            }
         }
 
         wss
             .clients
             .forEach((client) => {
                 if (jsond.app === client.app) {
-                    console.log('Broadcasting on app: %s', client.app);
+                    // console.log('Broadcasting on app: %s', client.app);
```
Comment (ColdSauce) [inline packages/hyperproxy-hub/src/index.js]: Remove commented out code

File: packages/hyperproxy-node/src/index.js [hunk 6]
```diff
@@ -35,33 +40,45 @@ export default class HyperproxyNode {
...
     */
-    async _handleData(client, key, {from, body, type}) {
+    async _handleData(key, {from, body, type}) {
         if (type === HUB_MSG_TYPE.RESPONSE) {
             return;
         }
 
         if (type === HUB_MSG_TYPE.REQUEST) {
+
+            console.log("DDDDDDDDDDDDDDDDDDD");
```
Comment (ColdSauce) [inline packages/hyperproxy-node/src/index.js]: Remove this or convert it to logging framework logs.

File: packages/hyperproxy-node/src/index.js [hunk 7]
```diff
@@ -35,33 +40,45 @@ export default class HyperproxyNode {
...
-                this._handleData(client, key, data);
+                this._handleData(key, data);
             } catch (e) {
                 console.error("ERROR in hyperproxy-node! ", err);
             }
         });
     }
 
+    _getPeerFrom(peerKey, peers) {
+        // peers.filter(peer => peer. == peerkKey)
```
Comment (ColdSauce) [inline packages/hyperproxy-node/src/index.js]: Remove commented out code? it seems like this function doesn't do anything. Should it still be here?

**Comment by ColdSauce** [comment]
That was a lot of merge conflicts :p @louisgv can you review?

File: packages/hyperproxy-hub-client/src/index.js [hunk 8]
```diff
@@ -25,31 +23,4 @@ export default class HyperproxyHubClient {
...
     }
-
-    /*
-        Handle swarm event reporting
-    */
-    handleSwarmEvent() {
-        const sw = this.swarm;
-
-        /* Swarm Event */
-        sw.on('close', function(e) {
```
Comment (louisgv) [inline packages/hyperproxy-hub-client/src/index.js]: We need this

File: packages/hyperproxy-hub-client/src/index.js [hunk 9]
```diff
@@ -25,31 +23,4 @@ export default class HyperproxyHubClient {
             hub.once(eventName, resolve);
         });
     }
-
-    /*
-        Handle swarm event reporting
-    */
-    handleSwarmEvent() {
```
Comment (louisgv) [inline packages/hyperproxy-hub-client/src/index.js]: We should keep this. Logging in experimental application is better than no log. It's P2P anyway :p

File: packages/hyperproxy-hub/src/index.js [hunk 10]
```diff
@@ -28,24 +29,49 @@ function onConnection(ws) {
...
+                } else {
+                    datMap[jsond.channel][jsond.message.from] = true;
+                    return;
+                }
             }
 
-            return;
-        }
+            if (jsond.message.type === HUB_MSG_TYPE.LEAVE) {
+                console.log('received a LEAVE event');
```
Comment (louisgv) [inline packages/hyperproxy-hub/src/index.js]: Change to use our log object

**Comment by ColdSauce** [comment]
@louisgv I'll take a look when I'm free tomorrow.

File: packages/hyperproxy-hub-client/src/index.js (hunk 9 above)
Comment (ColdSauce) [reply packages/hyperproxy-hub-client/src/index.js]: Ok makes sense! Could you explain your last sentence?

**Comment by ColdSauce** [comment]
@louisgv take a look when you get the chance

**Pull request merged**